app/geo/gui/src/ts/twxplore/gui/geo/reducerFunctions/updateAttributeStatesOfFeatureType.ts

```diff
@@ -0,0 +1,50 @@
+import {MapFeatureAttributeState} from "../states/map/MapFeatureAttributeState";
+import {MapFeature} from "../states/map/MapFeature";
+import {getFeatureAttributeByName} from "../attributeStrategies/getFeatureAttributeByName";
+
+export const updateAttributeStatesOfFeatureType = (
+  attributeStatesOfFeatureType: {
+    [featureAttributeName: string]: MapFeatureAttributeState;
+  },
+  addedFeature: MapFeature
+) => {
+  //Loop through the feature's attribute
+  for (const attribute of Object.keys(addedFeature)) {
+    //If the arttribute is numeric
+    if (getFeatureAttributeByName(attribute).isNumeric) {
+      //If this is the first time coming across this attribute for this feature type
+      if (
+        !attributeStatesOfFeatureType[attribute].min &&
```

Comment:
Duplicate expressions?

Also, min could be 0. You probably mean to check typeof(attributeStatesOfFeatureType[attribute].min) === "undefined".